build: add BranchReleaseSeries

RaduBerinde opened the pull request:
This change adds `build.BranchReleaseSeries()` which returns the major and minor in `version.txt`. This will be used to know the current release series when the latest `clusterversion` is not finalized.

We also clean up the code a bit: we separate the variables that are overridden by Bazel, and we use a different variable for the testing override (to make things more clear).

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/113827)
<!-- Reviewable:end -->

RaduBerinde committed:
build: add BranchReleaseSeries

This change adds `build.BranchReleaseSeries()` which returns the major
and minor in `version.txt`. This will be used to know the current
release series when the latest `clusterversion` is not finalized.

We also clean up the code a bit: we separate the variables that are
overridden by Bazel, and we use a different variable for the testing
override (to make things more clear).

Epic: none
Release note: None

RaduBerinde reviewed (commented):
TFTR!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/113827#-:-Nia-s1k323udn-BVtxh:b-814g8k)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @renatolabs and @rickystewart)

___
*[`pkg/build/info.go` line 208 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/113827#gh-1383598663:-Nia-t1X3Izr6_UQuL55:bknlrs0) ([raw file](https://github.com/cockroachdb/cockroach/blob/ce81f777918f236240012a239f460117c8443894/pkg/build/info.go#L208)):*
<details><summary><i>Previously, rickystewart (Ricky Stewart) wrote…</i></summary><blockquote>

How about adding the two conflicting versions in the error message to make it more readable?
</blockquote></details>

I made it allow "nested" uses like before.


<!-- Sent from Reviewable.io -->

RaduBerinde commented:
bors r+
